typescript/sdk/src/deploy/HyperlaneDeployer.ts

```diff
@@ -127,8 +127,11 @@ export abstract class HyperlaneDeployer<
     deployOpts?: DeployOptions,
   ): Promise<ReturnType<F['deploy']>> {
     const cachedContract = this.deployedContracts[chain]?.[contractName];
-    if (cachedContract) {
-      this.logger(`Recovered contract ${contractName} on ${chain}`);
+    if (cachedContract && !(cachedContract instanceof ProxiedContract)) {
```

Comment:
ditto with all this stuff to make it so that if an implementation address is deleted, the proxy is recovered but a new implementation is deployed